src/datastore/Restaurant.cpp

```diff
@@ -0,0 +1,270 @@
+/**
+ * CS1D - Fast Food Associates
+ *
+ * Implemention of the Restaurant class that provides a means
+ * for validating restaurants, string menu items and distances
+ * to other restaurants
+ *
+ * @author   edt
+ */
+
+
+#include "Restaurant.h"
+#include <string>
+#include <ostream>
+#include <iostream>
+
+int Restaurant::sm_NextNumber = 1;
+
+// Constructor - New Restaurant
+Restaurant::Restaurant(const string &Name, const float &Distance)
+        : m_Name(Name), m_fDistToSaddleback(Distance)
+{
+
+    m_fTotalPurchases = 0.0f;
+    m_nTotalCustomers = 0;
+    m_nUsedByCount = 0;
+    m_bDeleted = false;
+
+    m_nNumber =  sm_NextNumber++;
+
+    m_bInitialized = true;
+};
+
+// Constructor implementation
+Restaurant::Restaurant(
+         int        nNumber,
+         string     &Name,
+         float      &fDistToSaddleback,
+         float      &fTotalPurchases,
+         int        nTotalCustomers,
+         int        nUsedByCount,
+         bool       bDeleted,
+         vector<RestaurantDistance> &Distances,
+         vector<MenuItem>           &MenuItems,
+         vector<int>                &RefByTrips)
+: m_nNumber(nNumber), m_Name(Name), m_fDistToSaddleback(fDistToSaddleback),
+  m_fTotalPurchases(fTotalPurchases), m_nTotalCustomers(nTotalCustomers),
+  m_nUsedByCount(nUsedByCount), m_bDeleted(bDeleted),
+  m_Distances(Distances), m_Menu(MenuItems), m_ReferencedByTrips(RefByTrips)
+{
+    if (m_nNumber >= sm_NextNumber)
+    {
+        sm_NextNumber = m_nNumber + 1;
+    }
+    m_bInitialized = true;
+}
+
+// Destructor implementation
+Restaurant::~Restaurant()
+{
+}
+
+Restaurant::Restaurant(const Restaurant& src)
+: m_nNumber(src.m_nNumber), m_Name(src.m_Name), m_fDistToSaddleback(src.m_fDistToSaddleback),
+  m_fTotalPurchases(src.m_fTotalPurchases), m_nTotalCustomers(src.m_nTotalCustomers),
+  m_nUsedByCount(src.m_nUsedByCount), m_bDeleted(src.m_bDeleted),
+  m_Distances(src.m_Distances), m_Menu(src.m_Menu), m_ReferencedByTrips(src.m_ReferencedByTrips)
+{
+    m_bInitialized      = true;
+}
+
+Restaurant& Restaurant::operator=(const Restaurant& rhs)
+{
+   if (this == &rhs) {

```

Comment:
Braces should be on it's own line

```suggestion
   if (this == &rhs)
   {
```